dev/io.openliberty.microprofile.telemetry.2.0.internal/src/io/openliberty/microprofile/telemetry20/internal/config/OpenTelemetryVersionedConfigurationImpl.java

```diff
@@ -9,36 +9,127 @@
+
+        // Register observers for runtime metrics
+        Classes.registerObservers(openTelemetry);
+        Cpu.registerObservers(openTelemetry);
+        MemoryPools.registerObservers(openTelemetry);
+        Threads.registerObservers(openTelemetry);
+        GarbageCollector.registerObservers(openTelemetry);
+
+        return openTelemetry;
+
     }
 
     // Version specific default properties
-    @Override
-    public Map<String, String> getTelemetryPropertyDefaults() {
+    private Map<String, String> getTelemetryPropertyDefaults() {
         Map<String, String> telemetryProperties = new HashMap<String, String>();
-        //Logs are disabled by default, metrics is not
-        telemetryProperties.put(OpenTelemetryConstants.CONFIG_LOGS_EXPORTER_PROPERTY, "none");
-        telemetryProperties.put(OpenTelemetryConstants.ENV_LOGS_EXPORTER_PROPERTY, "none");
         return telemetryProperties;
     }
+
+    @Override
+    public OpenTelemetry createServerOpenTelemetryInfo(HashMap<String, String> telemetryProperties) {
+        try {
+            String otelInstanceName = OpenTelemetryConstants.OTEL_RUNTIME_INSTANCE_NAME;
+
+            if (AgentDetection.isAgentActive()) {
+                // If we're using the agent, it will have set GlobalOpenTelemetry and we must use its instance
+                // all config is handled by the agent in this case
+                return GlobalOpenTelemetry.get();
+            }
+
+            ClassLoader classLoader = OpenTelemetry.noop().getClass().getClassLoader();
+
+            //Builds tracer provider if user has enabled tracing aspects with config properties
+            if (!checkDisabled(telemetryProperties)) {
+                OpenTelemetry openTelemetry = AccessController.doPrivileged((PrivilegedAction<OpenTelemetry>) () -> {
+                    return buildOpenTelemetry(telemetryProperties, OpenTelemetryVersionedConfigurationImpl::customizeResource, classLoader);
+                });
+
+                if (openTelemetry != null) {
+                    return openTelemetry;
+                }
+            }
+
+            //By default, MicroProfile Telemetry tracing is off.
+            //The absence of an installed SDK is a “no-op” API
+            //Operations on a Tracer, or on Spans have no side effects and do nothing
+
+            Tr.info(tc, "CWMOT5100.tracing.is.disabled", otelInstanceName);
+
+            return null;
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CWMOT5002.telemetry.error", e));
+            return null;
+        }
+
+    }
+
+    private static Resource customizeResource(Resource resource, ConfigProperties c) {
+        ResourceBuilder builder = resource.toBuilder();
+        builder.put(AttributeKey.stringKey("service.name"), OpenTelemetryConstants.OTEL_RUNTIME_INSTANCE_NAME);
+
+        return builder.build();
+    }
+
+    private static boolean checkDisabled(Map<String, String> oTelConfigs) {
```

Comment:
In a follow up we should put this back in the common. Its here because the version neutral code caused an inexplicable OSGi error